### PR TITLE
Updated handling of VRF_VNI mapping and VLAN_VNI mapping for same VNI ID

### PR DESCRIPTION
**Pull request by tapashdas**
**What I did**
Updated handling of VRF_VNI mapping and VLAN_VNI mapping for same VNI ID

**Why I did it**
When a VNI (L3 VNI) is mapped to VRF,  corresponding VNI (L2 VNI) to Vlan mapping is not required.
Updated VxLAN Orchagent code to remove Vlan-VNI mapping entry when VRF is mapped to same VNI.
Also Re-add Vlan-VNI mapping entry when VRF-VNI mapping for same VNI is removed.

**How I verified it**

**Details if related**

### CODE REVIEW

**Comment by tapashdas** [comment]
@srj102 @RajeshPukhrajJain  Please review

**Comment by tapashdas** [comment]
@dgsudharsan Please review

**Comment by tapashdas** [comment]
@LaveenBrcm Please review

**Comment by tapashdas** [comment]
@hasan-brcm Please review

**Comment by srj102** [comment]
Please update the pytest as well.

**Review by dgsudharsan — CHANGES_REQUESTED** [review]
Please add UT.
There is a flow where VRF - VNI gets mapped first and then VLAN - VNI map arrives later. This PR doesn't handle it. Please handle that flow.

**Comment by tapashdas** [comment]
> 

Done

**Comment by dgsudharsan** [comment]
> > 
> 
> Done

Thanks. The existing UT fail due to new logic. Can you please update the UT?

**Comment by tapashdas** [comment]
Will do. Occupied with other work items.

**Comment by dgsudharsan** [comment]
> Will do. Occupied with other work items. Updated VRF-VNI test case. They are passing now. Current failures are from the newly added test case you added.

Understood. Can you please provide an ETA for addressing them?

**Review by dgsudharsan — APPROVED** [review]

**Comment by tapashdas** [comment]
‘/azpw run’

**Comment by tapashdas** [comment]
Please merge the changes.

File: orchagent/vxlanorch.cpp [hunk 1]
```diff
@@ -2256,6 +2288,32 @@ bool VxlanVrfMapOrch::delOperation(const Request& request)
...
+            SWSS_LOG_NOTICE("VxlanVrfMapOrch vni %d, isL2Vni %d\n", entry.vni_id, entry.isL2Vni);
+
+            if (entry.isL2Vni)
+            {
+                entry.vniVlanMapName = vniVlanMapName;
+                entry.vlan_id = vlan_id;
+                SWSS_LOG_DEBUG("add_tunnel_map_entry name %s, vlan %d, vni %d\n", entry.vniVlanMapName.c_str(), entry.vlan_id, entry.vni_id);
+            }
+        }
+        if(entry.isL2Vni)
```
Comment (prsunny) [inline orchagent/vxlanorch.cpp:2304]: please have space after` if `.

**Review by prsunny — APPROVED** [review]

**Comment by dgsudharsan** [comment]
@tapashdas Can you please raise submodule update?

**Comment by keboliu** [comment]
@StormLiangMS  would you please help to cherry-pick to the 202211 branch?

**Pull request merged**